urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/com/kokakiwi/mclauncher/utils/MCLogger.java

```diff
@@ -0,0 +1,191 @@
+                    global.info(s);
+                }
+                
+            });
+            
+            System.setErr(new PrintStream(System.err) {
+                
+                
+                public void print(String s)
+                {
+                    global.severe(s);
+                }
+                
+            });
+        }
+        catch (final IOException e)
+        {
+            e.printStackTrace();
+        }
+    }
+    
+    public static void debug(Object message)
+    {
+        global.log(DEBUG, message.toString());
+    }
+    
+    public static File getLauncherDir()
+    {
+        File dir = null;
+        
+        File root = new File(System.getProperty("user.home", ".") + "/");
+        final String appName = "mclauncher";
+        
+        switch (SystemUtils.getSystemOS())
+        {
+            case linux:
+            case solaris:
+                dir = new File(root, "." + appName + "/");
+                break;
+            
+            case macosx:
+                dir = new File(root, "Library/Application Support/" + appName);
+                break;
+            
+            case windows:
+                final String applicationData = System.getenv("APPDATA");
+                if (applicationData != null)
+                {
+                    root = new File(applicationData);
+                }
+                
+                dir = new File(root, '.' + appName + '/');
+                break;
+            
+            default:
+                dir = new File(root, appName + '/');
+                break;
+        }
+        
+        if (dir != null && !dir.exists() && !dir.mkdirs())
+        {
+            throw new RuntimeException(
+                    "The working directory could not be created: " + dir);
+        }
+        
+        return dir;
+    }
+    
+    public static class DebugLevel extends Level
+    {
+        private static final long serialVersionUID = 4164250706426396472L;
+        
+        protected DebugLevel()
+        {
+            super("DEBUG", Level.INFO.intValue() + 53);
+        }
+        
+    }
+    
+    public static class TerminalConsoleHandler extends ConsoleHandler
+    {
+        public TerminalConsoleHandler()
+        {
+            super();
+            setOutputStream(System.out);
+            setFormatter(new LogFormatter(false));
+        }
+    }
+    
+    public static class LogFormatter extends Formatter
+    {
+        private final static String LINE_DELIMITER = System.getProperty("line.separator");
+        
+        private final boolean       newLine;
+        
+        public LogFormatter(boolean newLine)
+        {
+            super();
+            this.newLine = newLine;
+        }
+        
+        
+        public String format(LogRecord record)
+        {
+            final StringBuffer sb = new StringBuffer();
+            
+            sb.append(new SimpleDateFormat("HH:mm:ss").format(new Date(record
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.